include/Flow.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Content: reads water flow sensor and creates packets, returns string and package
+ * Author : Ryan Leung
+ * Team   : Sustaingineering UBC
+ */
+ 
+#pragma once
+#include <Arduino.h>
+#include "LoRaTransceiver.h"
+#include "farmSensor.h"
+ 
+enum sensorType { digital, analog, counter };
+ 
+class flow : public farmSensor{
+ 
+private:
+    int m_pin;
+    String m_name;
+    String m_unit;
+    sensorType m_type;
+    char m_shortcut;
+    int m_data;
+    int readRaw();
+public:
+    flow(int pin, sensorType type, String name, String unit, char shortcut);
+    void initialize();
+    String read();
+    packet pack();
+};
+ 
+flow::flow(int pin, sensorType type, String name, String unit, char shortcut)
+{
+    m_pin = pin;
+    m_type = type;
+    m_name = name;
+    m_unit = unit;
+    m_shortcut = shortcut;
+    m_data = -1;
+}
+ 
+void flow::initialize()
+{
+    pinMode(m_pin, INPUT); //initializes digital pin as an input
+    attachInterrupt(digitalPinToInterrupt(m_pin),NbTopsCounterCaller, RISING); //and the interrupt is attached
+}
+ 
+//Calls counter function
+void NbTopsCounterCaller(){
+    NbTopsCounter();
+}
+ 
+//returns number of times that function is called
+int NbTopsCounter(){
```

Comment:
this function is declared in the global scope. However we only need it in the scope of this class. Therefore, it is better to make it a private memeber.